utils/dates.go

```diff
@@ -64,36 +65,40 @@ var isoFormats = []string{iso8601Format, iso8601NoSecondsFormat, iso8601Date}
 // ZeroDateTime is our uninitialized datetime value
 var ZeroDateTime = time.Time{}
 
-func dateFromFormats(env Environment, currentYear int, pattern *regexp.Regexp, d int, m int, y int, str string) (time.Time, error) {
+func dateFromFormats(env Environment, currentYear int, pattern *regexp.Regexp, d int, m int, y int, str string) (time.Time, string, error) {
 
-	matches := pattern.FindAllStringSubmatch(str, -1)
+	matches := pattern.FindAllStringSubmatchIndex(str, -1)
```

Comment:
previously time parsing happened on the same input as date parsing, but allowing single digits as hours means that the time regex matches against date components - so now we parse the date and return the remainder of the input - and only that is parsed for a time value